lib/internal/Magento/Framework/Data/AbstractSearchResult.php

```diff
@@ -59,9 +59,9 @@
     protected $entityFactory;
 
     /**
-     * @var \Magento\Framework\DB\Select
+     * @var \Magento\Framework\DB\Query
```

Comment:
Interface `\Magento\Framework\DB\QueryInterface` should be used instead of concrete class